src/xmipp/libraries/data/monogenic.cpp

```diff
@@ -0,0 +1,712 @@
+
+	transformer_inv.inverseFourierTransform(fftVRiesz_aux, VRiesz);
+
+	FOR_ALL_DIRECT_ELEMENTS_IN_MULTIDIMARRAY(amplitude)
+	{
+		DIRECT_MULTIDIM_ELEM(amplitude,n)+=DIRECT_MULTIDIM_ELEM(VRiesz,n)*DIRECT_MULTIDIM_ELEM(VRiesz,n);
+		DIRECT_MULTIDIM_ELEM(amplitude,n)=sqrt(DIRECT_MULTIDIM_ELEM(amplitude,n));
+	}
+
+
+
+	transformer_inv.FourierTransform(amplitude, fftVRiesz, false);
+
+	double raised_w = PI/(freqL-freq);
+	n=0;
+	FOR_ALL_DIRECT_ELEMENTS_IN_MULTIDIMARRAY(fftVRiesz)
+	{
+		double un=1.0/DIRECT_MULTIDIM_ELEM(iu,n);
+		if ((freqL)>=un && un>=freq)
+		{
+			DIRECT_MULTIDIM_ELEM(fftVRiesz,n) *= 0.5*(1 + cos(raised_w*(un-freq)));
+		}
+		else
+		{
+			if (un>freqL)
+			{
+				DIRECT_MULTIDIM_ELEM(fftVRiesz,n) = 0;
+			}
+		}
+	}
+
+	transformer_inv.inverseFourierTransform();
+//
+//	saveImg = amplitude;
+//	FileName iternumber;
+//	iternumber = formatString("_Amplitude_new_%i.vol", count);
+//	saveImg.write(fnDebug+iternumber);
+//	saveImg.clear();
+}
+
+
+// STATISTICSINBINARYMASK2: Estimates the staticstics of two maps:
+// Signal map "volS" and Noise map "volN". The signal statistics
+// are obtained by mean of a binary mask. The results are the mean
+// and variance of noise and signal, as well as the number of voxels
+// of signal and noise NS and NN respectively. The thr95 represents
+// the percentile 95 of noise distribution.
+void Monogenic::statisticsInBinaryMask2(const MultidimArray<double> &volS,
+		const MultidimArray<double> &volN,
+		MultidimArray<int> &mask, double &meanS, double &sdS2,
+		double &meanN, double &sdN2, double &significance, double &thr95, double &NS, double &NN)
+{
+	double sumS = 0, sumS2 = 0, sumN = 0, sumN2 = 0;
+	NN = 0;
+	NS = 0;
+	std::vector<double> noiseValues;
+
+	FOR_ALL_DIRECT_ELEMENTS_IN_MULTIDIMARRAY(volS)
+	{
+		if (DIRECT_MULTIDIM_ELEM(mask, n)>0)
+		{
+			double amplitudeValue=DIRECT_MULTIDIM_ELEM(volS, n);
+			sumS  += amplitudeValue;
+			sumS2 += amplitudeValue*amplitudeValue;
+			++NS;
+		}
+		if (DIRECT_MULTIDIM_ELEM(mask, n)>=0) //BE CAREFULL WITH THE =
+		{
+			double amplitudeValueN=DIRECT_MULTIDIM_ELEM(volN, n);
+			noiseValues.push_back(amplitudeValueN);
+			sumN  += amplitudeValueN;
+			sumN2 += amplitudeValueN*amplitudeValueN;
+			++NN;
+		}
+	}
+
+	std::sort(noiseValues.begin(),noiseValues.end());
+	thr95 = noiseValues[size_t(noiseValues.size()*significance)];
+	meanS = sumS/NS;
+	meanN = sumN/NN;
+	sdS2 = sumS2/NS - meanS*meanS;
+	sdN2 = sumN2/NN - meanN*meanN;
+}
+
+
+// STATISTICSINOUTBINARYMASK2: Estimates the staticstics of a single
+// map. The signal statistics are obtained by mean of a binary mask.
+// The results are the mean and variance of noise and signal, as well
+// as the number of voxels of signal and noise NS and NN respectively.
+// The thr95 represents the percentile 95 of noise distribution.
+void Monogenic::statisticsInOutBinaryMask2(const MultidimArray<double> &volS,
```

Comment:
duplicated code? I don't see any difference byt the mulitidyarray type

Reply:
There is a method that so something similar, but it doesnt provide statistics inside and outside. This does it